src/libasr/ASR.asdl

```diff
@@ -311,6 +311,8 @@ expr
     | DictLen(expr arg, ttype type, expr? value)
 
     | Var(symbol v)
+    | AddressOf(symbol v, ttype type)
```

Comment:
We have to be careful here.

This seems to be the same as CLoc. These two nodes should be merged then.

Also we want to be able to add an optional debug time checking for dangling pointers. So we have to be careful how these nodes are designed, so that it is possible to add checking.

Reply:
> This seems to be the same as CLoc (or GetPointer). These two nodes should be merged then.

`AddressOf` can be called on anything including array items, any type of variable, functions. However, `c_loc` can only be called on variables that too with `target` or `pointer` attribute. Also, `AddressOf` as a separate node will be very helpful when implementing pointer arithmetic features. Merging nodes can be option when we implement pointer arithmetic features, until then I would suggest not to merge nodes.

Reply:
I would not support pointer arithmetic, unless we absolutely have to. That's a feature that will be hard to check in Debug mode. We have to be extremely careful with introducing new features to ASR.

Reply:
See here: https://github.com/lfortran/lfortran/issues/3354.

I am afraid we'll have to revert this change.

Reply:
Okay I will revert this PR and turn off the test for `llvm`.

Reply:
Done in https://github.com/lcompilers/lc/pull/88